modules/dxp/apps/segments/segments-experiment-test/src/testIntegration/java/com/liferay/segments/experiment/web/internal/portlet/action/test/AddSegmentsExperimentMVCActionCommandTest.java

```diff
@@ -0,0 +1,279 @@
+
+		ReflectionTestUtil.setFieldValue(
+			PrefsPropsUtil.class, "_prefsProps", prefsPropsWrapper);
+
+		String description = RandomTestUtil.randomString();
+
+		SegmentsExperimentConstants.Goal goal =
+			SegmentsExperimentConstants.Goal.BOUNCE_RATE;
+
+		String name = RandomTestUtil.randomString();
+
+		String segmentsEntryName = RandomTestUtil.randomString();
+
+		SegmentsExperience segmentsExperience = _addSegmentsExperience(
+			segmentsEntryName);
+
+		MockLiferayPortletActionRequest mockLiferayPortletActionRequest =
+			_getMockLiferayPortletActionRequest(
+				description, goal.getLabel(), name, segmentsExperience);
+
+		try {
+			JSONObject jsonObject = ReflectionTestUtil.invoke(
+				_mvcActionCommand, "_addSegmentsExperiment",
+				new Class<?>[] {ActionRequest.class},
+				mockLiferayPortletActionRequest);
+
+			JSONObject segmentsExperimentJSONObject =
+				(JSONObject)jsonObject.get("segmentsExperiment");
+
+			Assert.assertEquals(
+				0.0, segmentsExperimentJSONObject.getDouble("confidenceLevel"),
+				0);
+			Assert.assertEquals(
+				description,
+				segmentsExperimentJSONObject.getString("description"));
+
+			SegmentsExperiment segmentsExperiment =
+				_segmentsExperimentLocalService.fetchSegmentsExperiment(
+					segmentsExperience.getSegmentsExperienceId(),
+					segmentsExperience.getClassNameId(),
+					segmentsExperience.getClassPK(),
+					new int[] {
+						SegmentsExperimentConstants.Status.DRAFT.getValue()
+					});
+
+			Assert.assertEquals(
+				liferayAnalyticsURL + "/tests/overview/" +
+					segmentsExperiment.getSegmentsExperimentKey(),
+				segmentsExperimentJSONObject.getString("detailsURL"));
+
+			Assert.assertTrue(
+				segmentsExperimentJSONObject.getBoolean("editable"));
+			Assert.assertEquals(
+				String.valueOf(
+					JSONUtil.put(
+						"label", "Bounce Rate"
+					).put(
+						"value", goal.getLabel()
+					)),
+				String.valueOf(
+					segmentsExperimentJSONObject.getJSONObject("goal")));
+			Assert.assertEquals(
+				name, segmentsExperimentJSONObject.getString("name"));
+			Assert.assertEquals(
+				segmentsEntryName,
+				segmentsExperimentJSONObject.getString("segmentsEntryName"));
+			Assert.assertEquals(
+				String.valueOf(segmentsExperience.getSegmentsExperienceId()),
+				segmentsExperimentJSONObject.getString("segmentsExperienceId"));
+			Assert.assertEquals(
+				String.valueOf(segmentsExperiment.getSegmentsExperimentId()),
+				segmentsExperimentJSONObject.getString("segmentsExperimentId"));
+			Assert.assertEquals(
+				String.valueOf(
+					JSONUtil.put(
+						"label", "Draft"
+					).put(
+						"value",
+						SegmentsExperimentConstants.Status.DRAFT.getValue()
+					)),
+				String.valueOf(
+					segmentsExperimentJSONObject.getJSONObject("status")));
+		}
+		finally {
+			ReflectionTestUtil.setFieldValue(
+				PrefsPropsUtil.class, "_prefsProps", prefsProps);
+		}
+	}
+
+	private SegmentsExperience _addSegmentsExperience(String segmentsEntryName)
```

Comment:
Ass segmentsEntryName is random, do we really need to define this parameter, could it not be obtained from the Segment Experience?

Reply:
Sure, but would need to retrieve the segments entry afterwards, so a new variable is needed:

```java
		SegmentsEntry segmentsEntry = _segmentsEntryLocalService.getSegmentsEntry(
			segmentsExperience.getSegmentsEntryId());
		
		String segmentsEntryName = segmentsEntry.getName();
```

I prefer to make the segmentsEntryName assignment explicit. But, let me know what you think.

Reply:
What I see is that we have a lot of string variables that have not a semantic meaning (goal, name, segmentEntryName) ...

Reply:
So it is very hard to understand, what description means in a real environment without taking a look into the code.

Reply:
not I get it, do you mean it would clearer to directly use random strings and retrieve the `SegmentsExperiment` after the call and assert with that object? I was tempted to do that, but It was a bit weird because that's exactly what I'm testing. But, I had to use that SegmentsExperiment to assert `detailsURL` property.

Reply:
What I want, for the user that reads this test in the future is for them to understand what is the meaning of every Request parameter in context. In the current test, it is really hard to understand what is the meaning of every variable and why should we fill it.

Reply:
I see your point, but how could I improve the readability? I mean, the command itself It's already difficult to understand:

com.liferay.segments.experiment.web.internal.portlet.action.AddSegmentsExperimentMVCActionCommand#_addSegmentsExperiment

```java
		SegmentsExperiment segmentsExperiment =
			_segmentsExperimentService.addSegmentsExperiment(
				ParamUtil.getLong(actionRequest, "segmentsExperienceId"),
				ParamUtil.getLong(actionRequest, "classNameId"),
				ParamUtil.getLong(actionRequest, "classPK"),
				ParamUtil.getString(actionRequest, "name"),
				ParamUtil.getString(actionRequest, "description"),
				ParamUtil.getString(actionRequest, "goal"),
				ParamUtil.getString(actionRequest, "goalTarget"),
				ServiceContextFactory.getInstance(actionRequest));
```

Reply:
I'm keeping an eye on `com.liferay.analytics.reports.test.util.MockContextUtil`

Reply:
Hey @darquesdev!!

I don't have a clear solution for this, I hope you, that have a better knowledge of the command could find a way to read easily what we are trying to do it (usually to use an object to represent what we try to accomplish should help).

As you point me to the code in AddSegmentsExperimentMVCActionCommand, the test could help us to refractor a hard-to-read code too.

It is up to you!!, for me, this test is very hard o read, but if you think is OK, and this can't be improved. You can forward the pull

Thanks!!